src/main/java/com/dataloom/datastore/search/controllers/SearchController.java

```diff
@@ -57,31 +59,38 @@
 public class SearchController implements SearchApi, AuthorizingComponent {
 
     @Inject
-    private SearchService searchService;
+    private SearchService          searchService;
 
     @Inject
-    private EdmService edm;
+    private EdmService             edm;
 
     @Inject
-    private AuditQueryService aqs;
+    private AuditQueryService      aqs;
 
     @Inject
-    private AuthorizationManager authorizations;
-    
+    private AuthorizationManager   authorizations;
+
     @Inject
     private EdmAuthorizationHelper authorizationsHelper;
 
     @RequestMapping(
-            path = { "/", "" },
-            method = RequestMethod.POST,
-            produces = { MediaType.APPLICATION_JSON_VALUE } )
-    public SearchResult executeEntitySetKeywordQuery( @RequestBody Search search ) {
+        path = { "/", "" },
+        method = RequestMethod.POST,
+        produces = { MediaType.APPLICATION_JSON_VALUE } )
+    public SearchResult executeEntitySetKeywordQuery(
+            @RequestBody Search search,
+            HttpServletResponse response ) throws IOException {
         if ( !search.getOptionalKeyword().isPresent() && !search.getOptionalEntityType().isPresent()
                 && !search.getOptionalPropertyTypes().isPresent() ) {
-            throw new BadRequestException(
-                    "You must specify at least one request body param (keyword 'kw', entity type id 'eid', or property type ids 'pid'" );
+            response.sendError( HttpServletResponse.SC_BAD_REQUEST,
```

Comment:
For consistency we probably want to throw an exception, and return the http response in `DatastoreExceptionHandler` instead.